### ISSUE TITLE
When recreating the pod, CiliumEgressGatewayPolicy is applied only after ~1 minutes. The new CiliumEgressGatewayPolicy is not applied until restart cilium agent on node.

### ISSUE THREAD

**Issue by kalx77** [cilium/cilium#26503]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Cluster based on AWS EKS.
Several gray IPs like addresses 10.80.X.X are attached to the node on the eth(0/1/2) interfaces. With these gray addresses are associated white Elastic IPs.
When creating a CiliumEgressGatewayPolicy and a pod, as in the example, the CiliumEgressGatewayPolicy is not applied until (we tried to wait about 20 minutes) until the cilium agent on the node is restarted (by restarting the pod).
If the CiliumEgressGatewayPolicy is already in the cluster and has been applied, then after restarting a pod that matches this policy, the policy is applied to it in about a minute.

```
root@ip-10-80-62-96:/home/cilium# cilium status
KVStore:                 Ok   Disabled
Kubernetes:              Ok   1.23+ (v1.23.17-eks-c12679a) [linux/amd64]
Kubernetes APIs:         ["cilium/v2::CiliumClusterwideNetworkPolicy", "cilium/v2::CiliumEgressGatewayPolicy", "cilium/v2::CiliumEgressNATPolicy", "cilium/v2::CiliumEndpoint", "cilium/v2::CiliumLocalRedirectPolicy", "cilium/v2::CiliumNetworkPolicy", "cilium/v2::CiliumNode", "core/v1::Namespace", "core/v1::Node", "core/v1::Pods", "core/v1::Service", "discovery/v1::EndpointSlice", "networking.k8s.io/v1::NetworkPolicy"]
KubeProxyReplacement:    Strict   [eth0 10.80.62.96, eth1 10.80.62.4, eth2 10.80.62.16]
Host firewall:           Disabled
CNI Chaining:            none
Cilium:                  Ok   1.12.5 (v1.12.5-701acde)
NodeMonitor:             Listening for events on 4 CPUs with 64x4096 of shared memory
Cilium health daemon:    Ok
IPAM:                    IPv4: 36/510 allocated from 10.85.26.0/23,
BandwidthManager:        Disabled
Host Routing:            Legacy
Masquerading:            BPF   [eth0, eth1, eth2]   10.85.26.0/23 [IPv4: Enabled, IPv6: Disabled]
Controller Status:       188/188 healthy
Proxy Status:            No managed proxy redirect
Global Identity Range:   min 256, max 65535
Hubble:                  Ok   Current/Max Flows: 4095/4095 (100.00%), Flows/s: 274.21   Metrics: Disabled
Encryption:              Disabled
Cluster health:          36/36 reachable   (2023-06-27T10:47:24Z)
```

```
Cilium config:
  agent-not-ready-taint-key: node.cilium.io/agent-not-ready
  arping-refresh-period: 30s
  auto-direct-node-routes: "false"
  bpf-lb-external-clusterip: "false"
  bpf-lb-map-max: "65536"
  bpf-lb-sock: "false"
  bpf-map-dynamic-size-ratio: "0.0025"
  bpf-policy-map-max: "16384"
  bpf-root: /sys/fs/bpf
  cgroup-root: /run/cilium/cgroupv2
  cilium-endpoint-gc-interval: 5m0s
  cluster-id: "0"
  cluster-name: default
  cluster-pool-ipv4-cidr: 10.85.0.0/16
  cluster-pool-ipv4-mask-size: "23"
  custom-cni-conf: "false"
  debug: "false"
  devices: eth+
  disable-cnp-status-updates: "true"
  disable-endpoint-crd: "false"
  enable-auto-protect-node-port-range: "true"
  enable-bgp-control-plane: "false"
  enable-bpf-clock-probe: "true"
  enable-bpf-masquerade: "true"
  enable-endpoint-health-checking: "true"
  enable-endpoint-routes: "true"
  enable-health-check-nodeport: "true"
  enable-health-checking: "true"
  enable-hubble: "true"
  enable-ipv4: "true"
  enable-ipv4-egress-gateway: "true"
  enable-ipv4-masquerade: "true"
  enable-ipv6: "false"
  enable-ipv6-masquerade: "true"
  enable-k8s-terminating-endpoint: "true"
  enable-l2-neigh-discovery: "true"
  enable-l7-proxy: "false"
  enable-local-redirect-policy: "true"
  enable-policy: default
  enable-remote-node-identity: "true"
  enable-svc-source-range-check: "true"
  enable-vtep: "false"
  enable-well-known-identities: "false"
  enable-xt-socket-fallback: "true"
  hubble-disable-tls: "false"
  hubble-listen-address: :4244
  hubble-socket-path: /var/run/cilium/hubble.sock
  hubble-tls-cert-file: /var/lib/cilium/tls/hubble/server.crt
  hubble-tls-client-ca-files: /var/lib/cilium/tls/hubble/client-ca.crt
  hubble-tls-key-file: /var/lib/cilium/tls/hubble/server.key
  identity-allocation-mode: crd
  install-egress-gateway-routes: "true"
  install-iptables-rules: "true"
  install-no-conntrack-iptables-rules: "false"
  ipam: cluster-pool
  kube-proxy-replacement: strict
  kube-proxy-replacement-healthz-bind-address: ""
  monitor-aggregation: medium
  monitor-aggregation-flags: all
  monitor-aggregation-interval: 5s
  node-port-bind-protection: "true"
  nodes-gc-interval: 5m0s
  operator-api-serve-addr: 127.0.0.1:9234
  preallocate-bpf-maps: "false"
  procfs: /host/proc
  remove-cilium-node-taints: "true"
  set-cilium-is-up-condition: "true"
  sidecar-istio-proxy-image: cilium/istio_proxy
  skip-cnp-status-startup-clean: "false"
  synchronize-k8s-nodes: "true"
  tofqdns-dns-reject-response-code: refused
  tofqdns-enable-dns-compression: "true"
  tofqdns-endpoint-max-ip-per-hostname: "50"
  tofqdns-idle-connection-grace-period: 0s
  tofqdns-max-deferred-connection-deletes: "10000"
  tofqdns-min-ttl: "3600"
  tofqdns-proxy-response-max-delay: 100ms
  tunnel: vxlan
  unmanaged-pod-watcher-interval: "15"
  vtep-cidr: ""
  vtep-endpoint: ""
  vtep-mac: ""
  vtep-mask: ""
```

Policy and pod example:
```
apiVersion: cilium.io/v2
kind: CiliumEgressGatewayPolicy
metadata:
  name: egress-test
spec:
  destinationCIDRs:
  - 0.0.0.0/5
  - 8.0.0.0/7
  - 11.0.0.0/8
  - 12.0.0.0/6
  - 16.0.0.0/4
  - 32.0.0.0/3
  - 64.0.0.0/2
  - 128.0.0.0/3
  - 160.0.0.0/5
  - 168.0.0.0/6
  - 172.0.0.0/12
  - 172.32.0.0/11
  - 172.64.0.0/10
  - 172.128.0.0/9
  - 173.0.0.0/8
  - 174.0.0.0/7
  - 176.0.0.0/4
  - 192.0.0.0/9
  - 192.128.0.0/11
  - 192.160.0.0/13
  - 192.169.0.0/16
  - 192.170.0.0/15
  - 192.172.0.0/14
  - 192.176.0.0/12
  - 192.192.0.0/10
  - 193.0.0.0/8
  - 194.0.0.0/7
  - 196.0.0.0/6
  - 200.0.0.0/5
  - 208.0.0.0/4
  egressGateway:
    egressIP: 10.80.62.62
    nodeSelector:
      matchLabels:
        kubernetes.io/hostname: ip-10-80-62-96.eu-central-1.compute.internal
  selectors:
  - podSelector:
      matchLabels:
        app: test
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: ergess-test-pod
  labels:
    app: test
spec:
  selector:
    matchLabels:
      app: test
  replicas: 1
  template:
    metadata:
      labels:
        app: test
    spec:
      containers:
      - name: netshoot
        image: nicolaka/netshoot
        command: ["/bin/bash"]
        args: ["-c", "while true; do sleep 10; done"]
      nodeName: ip-10-80-62-96.eu-central-1.compute.internal
```

### Cilium Version

Client: 1.12.5 701acde 2022-12-15T16:03:30-08:00 go version go1.18.9 linux/amd64
Daemon: 1.12.5 701acde 2022-12-15T16:03:30-08:00 go version go1.18.9 linux/amd64

### Kernel Version

Linux version 5.4.228-131.415.amzn2.x86_64 (mockbuild@ip-10-0-35-227) (gcc version 7.3.1 20180712 (Red Hat 7.3.1-15) (GCC)) #1 SMP Tue Dec 20 12:51:02 UTC 2022

### Kubernetes Version

Server Version: version.Info{Major:"1", Minor:"23+", GitVersion:"v1.23.17-eks-c12679a", GitCommit:"d5ce2cee85d99653d6f8c278043213db21b1cd72", GitTreeState:"clean", BuildDate:"2023-05-22T20:32:28Z", GoVersion:"go1.19.6", Compiler:"gc", Platform:"linux/amd64"}

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jibi** [comment]
Hi @kalx77 :wave: 

are you able to reproduce this even with the latest 1.12 release (so `v1.12.11`)?

**Comment by kalx77** [comment]
> Hi @kalx77 wave
> 
> are you able to reproduce this even with the latest 1.12 release (so `v1.12.11`)?

Hi @jibi :wave: 

Now updated to v1.12.11 - the situation has not changed
```
root@ip-10-80-62-96:/home/cilium# cilium version
Client: 1.12.11 6511094 2023-06-14T03:44:07+00:00 go version go1.18.10 linux/amd64
Daemon: 1.12.11 6511094 2023-06-14T03:44:07+00:00 go version go1.18.10 linux/amd64
```

**Comment by jibi** [comment]
Could you enable `debug` in the configmap and collect a sysdump of the cluster please?

**Comment by kalx77** [comment]
Collected 2 dumps. What did you do:
1. Restart cilium pod on node ip-10-80-62-39.eu-central-1.compute.internal. New pod was created with name cilium-qrcr7
2. Started dump collection `cilium sysdump --debug kube-system`
3. Wait until the dump is collected [cilium-sysdump-20230628-172342.zip](https://mega.nz/file/BydTBLrB#HWSQ9TfpSnMqz5QyPsRk0jPSywT7PslQmX7mB-Yp4XA)

After that
4. Restart cilium pod on node ip-10-80-62-39.eu-central-1.compute.internal. New pod was created with name cilium-j7ztn
5. Started dump collection `cilium sysdump --debug kube-system`
6. Delete test pod ergess-test-pod
7. Entered to the new pod ergess-test-pod and ran `curl ipinfo.io` about once every 5 seconds until the policy was applied (Approximately 1 minute)
8. Wait until the dump is collected [cilium-sysdump-20230628-173657.zip](https://mega.nz/file/hu1TRYxL#Rwgg4sCHwu_8vX1JbP3XKx7VcJz_x1MzV1BcKdezCzM)

**Comment by kalx77** [comment]
We gathered our strength and updated the cilium in the cluster to 1.13.4. The problem is gone, so we can close the issue. Thanks for help!=)

**Comment by julianwiedmann** [comment]
> We gathered our strength and updated the cilium in the cluster to 1.13.4. The problem is gone, so we can close the issue. Thanks for help!=)

Awesome, thank you for the update!
